pkg/services/sqlstore/migrations/dashboard_public_config_mig.go

```diff
@@ -0,0 +1,26 @@
+package migrations
+
+import (
+	. "github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addPublicDashboardMigration(mg *Migrator) {
+	var dashboardPublicCfgV1 = Table{
+		Name: "dashboard_public_config",
+		Columns: []*Column{
+			{Name: "uid", Type: DB_BigInt, IsPrimaryKey: true},
+			{Name: "dashboard_uid", Type: DB_NVarchar, Nullable: false},
```

Comment:
```suggestion
			{Name: "dashboard_uid", Type: DB_NVarchar, Length: 40, Nullable: false},
```
You need to specify the length otherwise MySQL results to an invalid query:
```
ERROR[05-02|11:59:32] Exec failed                              logger=migrator error="Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL\n, `org_id` BIGINT(20) ' at line 3" sql="CREATE TABLE IF NOT EXISTS `dashboard_public_config` (\n`uid` BIGINT(20) PRIMARY KEY NOT NULL\n, `dashboard_uid` VARCHAR CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL\n, `org_id` BIGINT(20) NOT NULL\n, `refresh_rate` INT NOT NULL DEFAULT 30\n, `template_variables` MEDIUMTEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NULL\n, `time_variables` TEXT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci NOT NULL\n) ENGINE=InnoDB DEFAULT CHARSET utf8mb4 COLLATE utf8mb4_unicode_ci;"
```

Reply:
Should't integration tests catch this :thinking:

Reply:
The table creating is hidden behind the feature flag so CI tests won't catch this unless they specifically set it. Which leads us to another question: do we want to keep doing this or creating the table no matter the feature is enabled or not and having in that way more consistent/deterministic migrations?

Reply:
We had a similar discussion after @dafydd-t's comments here https://github.com/grafana/grafana/pull/47709#discussion_r858952114 and in the linked PR, and ended up dropping the `ifs`

Reply:
@bergquist @ArturWierzbicki @papagian this seems like something worth documenting in our[ contribution docs](https://github.com/grafana/grafana/blob/main/contribute/architecture/backend/database.md) and perhaps leaving a comment in migrations.go